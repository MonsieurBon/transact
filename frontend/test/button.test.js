import { fixture } from '@open-wc/testing-helpers';
import Button from '../src/components/button.js';

describe('Button', () => {
  beforeAll(() => {
    window.customElements.define('t-button', Button);
  });

  it('should render', async () => {
    const el = await fixture('<t-button>My button</t-button>');
    expect(el.shadowRoot.innerHTML).toMatchSnapshot();
  });
});
