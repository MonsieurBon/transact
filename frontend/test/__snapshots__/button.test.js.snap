// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
"
<style>
  :host {
    contain: content;
  }
  
  button {
    background: transparent;
    border: 1px solid black;
    border-radius: 5px;
    height: 35px;
    padding: 0 20px;
  }
  
  :host(.dark) button {
    background: black;
    border-color: white;
    color: white;
  }
  
  :host(.light) button {
    background: white;
  }
</style>
<button><slot></slot></button>
"
`;
